pysap/base/transform.py

```diff
@@ -233,8 +244,16 @@ def _set_data(self, data):
             print("[info] Replacing existing input data array.")
         if not all([e == data.shape[0] for e in data.shape]):
             raise ValueError("Expect a square shape data.")
-        if data.ndim != 2:
-            raise ValueError("Expect a two-dim data array.")
+        if data.ndim != self.data_dim:
+            if self.data_dim == 2:
+                raise ValueError("This wavelet can only be applied on 2D"
+                                 " square images")
```

Comment:
You don't test explicitelly the image shape.
Is the error message appropriate?
On top of that why having a two level test?
Just test it directly as it was done previously and customize the error message using .format(self.data_dim).

Reply:
the data.ndim is the shape of the image, (more exactly its number of dim)
I corrected the two level test by using format